File: Programmers/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> map = new HashMap<>();
        for (String name : participant)
            map.put(name, map.getOrDefault(name, 0) + 1);

        for (String name : completion) {
            if (map.get(name) > 1) map.put(name, map.get(name) - 1);
            else map.remove(name);
        }
        answer = map.keySet().iterator().next();
        return answer;
    }
}

File: Programmers/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    map<string, int> m;
    for (string name : participant) m[name]++;
    for (string name : completion)
        if (m[name] > 1) m[name]--;
        else m.erase(name);
    for (auto it : m) {
        answer = it.first;
        break;
    }
    return answer;
}